lib/omnibus/fetchers/git_fetcher.rb

```diff
@@ -18,12 +18,12 @@ module Omnibus
   class GitFetcher < Fetcher
     #
     # A fetch is required if the git repository is not cloned or if the local
-    # revision does not match the desired revision.
+    # revision does not match the desired revision or if we always fetch tags
     #
     # @return [true, false]
     #
     def fetch_required?
-      !(cloned? && contains_revision?(resolved_version))
+      !(cloned? && contains_revision?(resolved_version) && !always_fetch_tags?)
```

Comment:
Nit: !(cloned? && contains_revision?(resolved_version)) || always_fetch_tags? might be more readable.